### PR TITLE
:sparkles: merge acunetix and acunetix360

### PR DESCRIPTION
**Pull request by manuel-sommer**

### CODE REVIEW

**Comment by manuel-sommer** [comment]
Ready to review @mtesauro

**Review by mtesauro — APPROVED** [review]
Approved

**Review by grendel513 — APPROVED** [review]

**Comment by manuel-sommer** [comment]
friendly reminder @cneill and @Maffooch

**Review by Maffooch — APPROVED** [review]

File: dojo/db_migrations/0204_merge_acunetix.py
Comment (blakeaowens) [inline dojo/db_migrations/0204_merge_acunetix.py:54]: Would you mind adding a reverse operation to this migration so it can easily be reversed?

Comment (manuel-sommer) [reply dojo/db_migrations/0204_merge_acunetix.py:54]: How can I add a reverse operation? Do you have an example?

File: dojo/tools/acunetix/parser.py
Comment (cneill) [inline dojo/tools/acunetix/parser.py:29]: Could you separate the code in each of these blocks (`.xml` vs `.json`) into separate helper methods so it's easier to see which bits relate to which parser, something like `parse_acunetix_xml` and `parse_acunetix360_json` or similar? That way we have all the context for what each bit is meant to handle without tracking the whitespace through ~200 lines.

Comment (manuel-sommer) [reply dojo/tools/acunetix/parser.py:29]: Good idea, I can do that.

Comment (manuel-sommer) [reply dojo/tools/acunetix/parser.py:21]: Done @cneill

File: dojo/db_migrations/0204_merge_acunetix.py
Comment (manuel-sommer) [reply dojo/db_migrations/0204_merge_acunetix.py]: @blakeaowens: I looked into it. I have questions regarding the implementation as I don't have experience with building db migrations:
- Let's assume you have now merged Acunetix360 findings with Acunetix findings to Acunetix findings. How can I now distinguish between the findings in order to build again Acunetix360 findings?
- Is this technically actually possible? Could you please take a look at my last commit?

**Review by devGregA — APPROVED** [review]

**Comment by manuel-sommer** [comment]
I reverted my last commit as I guess that a revert action for this db migration is technically not possible.

**Comment by manuel-sommer** [comment]
To me it is ready to get merged.

**Comment by manuel-sommer** [comment]
Lol, whats aoing on with the unittests.

**Comment by manuel-sommer** [comment]
Hi @mtesauro, I already have 4 approvals here, but what about this PR and https://github.com/DefectDojo/django-DefectDojo/discussions/9690 ?

**Comment by mtesauro** [comment]
@manuel-sommer About this PR and #9690 

From what I understand about Acunetix and Acunetix360, they are both DAST scanners from the same vendor with different file formats (XML and JSON).

So, I'd expect they are similar enough to have a combined parser and still be able to write a good dedup algorithm. When they were separated parsers, they had [matching dedup algorithms](https://github.com/DefectDojo/django-DefectDojo/pull/9522/files#diff-767da9c92328dd092f33c455f007b9c3a936854ccab95f8130cf821d18af82d0L1203) so I don't see the same problem as combining say a DAST, and SCA tools output if from the same vendor.

About the 4 approvals and no merge, I was waiting to hear back from @blakeaowens since he raised the question about migrations.

**Comment by manuel-sommer** [comment]
Friendly reminder @blakeaowens

**Comment by blakeaowens** [comment]
I am fine with no reverse-migration method for this PR.

**Comment by manuel-sommer** [comment]
Hi @mtesauro, fyi, I updated db migrations. It would be nice if we could merge this.

**Comment by mtesauro** [comment]
@manuel-sommer Sure thing. Closed and opened this to try to get that Flake8 test happy. Once that's happy, we're good to merge.

**Comment by ekondur** [comment]
Hi @manuel-sommer, 
We got some errors after this change, are you from the acx team? if so, can you pls slack me, (Emrah KONDUR)

**Comment by manuel-sommer** [comment]
No, I am not from the acunetix team. If you submit a new issue with sample findings, I can help you fix the problem

**Comment by ekondur** [comment]
Hi @manuel-sommer 
Different products and different customers. It will cause confusion. You can check it here,
https://www.acunetix.com/product/acunetix360/
https://www.acunetix.com/product/premium/
Can we pls add 360 one as well? 
Many of our customers are affected, can you help us do it faster, please? @mtesauro

**Comment by ekondur** [comment]
Hi @manuel-sommer 
I will try without the 360 suffix, if it works we can change it on our side (360 code base), 

Can you please inform the relevant vendor in advance about such changes? FYI @mtesauro

**Comment by ekondur** [comment]
Hi @manuel-sommer
We discussed it with the team and decided to use both Acunetix and Acunetix360, these are different products and have different customers. It might be confusing because our customers have used these products as Acunetix Premium and Acunetix360 for many years. Even the teams in charge of products are different. 
https://www.acunetix.com/product/acunetix360/
https://www.acunetix.com/product/premium/
Also, we have concerns about if one of the report models is changed, we should also consider the compatibility of the other. So it would be better to use them separately. 
Could you please revert this change?

**Comment by ekondur** [comment]
Kindly remind @mtesauro, it is a bit urgent, it affected many of our customers.

**Comment by ekondur** [comment]
Hi @devGregA I'm sorry, I have no idea why there was no receive. Can you please inform us about the process so we can revert it and publish it as soon as possible?

**Comment by ekondur** [comment]
Any update @manuel-sommer @devGregA @mtesauro?

**Pull request merged**
